Lib/test/test_base64.py

```diff
@@ -351,6 +351,81 @@ def test_b32decode_error(self):
                 with self.assertRaises(binascii.Error):
                     base64.b32decode(data.decode('ascii'))
 
+    def test_b32hexencode(self):
+        eq = self.assertEqual
+        eq(base64.b32hexencode(b''), b'')
+        eq(base64.b32hexencode(b'\x00'), b'00======')
+        eq(base64.b32hexencode(b'a'), b'C4======')
+        eq(base64.b32hexencode(b'ab'), b'C5H0====')
+        eq(base64.b32hexencode(b'abc'), b'C5H66===')
+        eq(base64.b32hexencode(b'abcd'), b'C5H66P0=')
+        eq(base64.b32hexencode(b'abcde'), b'C5H66P35')
+        # Non-bytes
+        self.check_other_types(base64.b32hexencode, b'abcd', b'C5H66P0=')
+        self.check_encode_type_errors(base64.b32hexencode)
+
+    def test_b32hexdecode(self):
+        eq = self.assertEqual
+        tests = {b'': b'',
+                 b'00======': b'\x00',
+                 b'C4======': b'a',
+                 b'C5H0====': b'ab',
+                 b'C5H66===': b'abc',
+                 b'C5H66P0=': b'abcd',
+                 b'C5H66P35': b'abcde',
+                 }
+        for data, res in tests.items():
+            eq(base64.b32hexdecode(data), res)
+            eq(base64.b32hexdecode(data.decode('ascii')), res)
+        # Non-bytes
+        self.check_other_types(base64.b32hexdecode, b'C5H66===', b"abc")
+        self.check_decode_type_errors(base64.b32hexdecode)
+
+    def test_b32hexdecode_casefold(self):
+        eq = self.assertEqual
+        tests = {b'': b'',
+                 b'00======': b'\x00',
+                 b'C4======': b'a',
+                 b'C5H0====': b'ab',
+                 b'C5H66===': b'abc',
+                 b'C5H66P0=': b'abcd',
+                 b'C5H66P35': b'abcde',
+                 # Lower cases
+                 b'c4======': b'a',
+                 b'c5h0====': b'ab',
+                 b'c5h66===': b'abc',
+                 b'c5h66p0=': b'abcd',
+                 b'c5h66p35': b'abcde',
+                 }
+
+        for data, res in tests.items():
```

Comment:
Per the other comment, I'd make these subtests as well.

In my other comment I used a list of tuples instead of a dictionary, but I think a dictionary is fine (at least in Python 3.6+), so I'd recommend picking whichever one you like and using it in both.